tests/jeos/check_efi_setup.pm

```diff
@@ -0,0 +1,230 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: Check EFI boot in JeOS images
+#   1) Get gpt drive and find ESP
+#   2) Check ESP partition alignment
+#   3) Perform fsck on ESP
+#   4) Check EFI settings after firstboot
+#   5) Update grub.cfg and reboot
+#   6) Check EFI settings after grub.cfg update
+#   7) Reinstall shim and reboot
+#   8) Check EFI settings after shim reinstal
+#   9) Reinitialize bootloader using pbl
+#   10) Check EFI settings after reinit
+# Maintainer: Martin Loviska <mloviska@suse.com>
+
+use strict;
+use warnings;
+use base "opensusebasetest";
+use testapi;
+use utils 'zypper_call';
+use version_utils qw(is_leap is_opensuse);
+use jeos;
+use constant {
+    SYSFS_EFI_BITS      => '/sys/firmware/efi/fw_platform_size',
+    FSTAB               => '/etc/fstab',
+    GRUB_DEFAULT        => '/etc/default/grub',
+    GRUB_CFG            => '/boot/grub2/grub.cfg',
+    SYSCONFIG_BOOTLADER => '/etc/sysconfig/bootloader'
+};
+
+sub efibootmgr_current_boot {
+    # helper function to retrieve location of efi executable
+    my $get_efi_exec = sub {
+        my $str = shift;
+        $str or die "Empty record in efibootmgr parser!\n";
+        $str =~ s/\\/\//g;
+        return substr $str, index($str, '(') + 1, (index($str, ')') - 1) - index($str, '(');
+    };
+
+    my @ebm_raw = split /\n/, script_output 'efibootmgr --verbose';
+    die "EFI variables are not supported\n" if ($ebm_raw[0] =~ /EFI variables are not supported/);
+
+    # get current boot no., the last element of the first line
+    my $h->{bootid} = (split /\s+/, shift @ebm_raw)[-1];
+    ($h->{bootid}) or die "No id of current boot found!\n";
+
+    # booted efi details
+    my @boot_entry_tokens = split /\s/, (grep /^boot$h->{bootid}\*.*$/i, @ebm_raw)[0];
+    # remove Boot{$boot_no}
+    shift @boot_entry_tokens;
```

Comment:
maybe it make sense to log what we are removing , just in case format will change at some point so we will have chance to notice this 
```suggestion
    recod_info('removing Boot{$boot_no} from efibootmgr --verbose output' . shift @boot_entry_tokens);
```

or less verbose version : 
```
    diag('removing Boot{$boot_no} from efibootmgr --verbose output' . shift @boot_entry_tokens);

Reply:
Mainly we do not have serial terminal on each backend. It would be harder to get this information from the screenshots. In the erial_terminal.txt it would be visible if you would compare few lines.

Was not sure how to approach this properly, I am glad that *diag* is sufficient for others as well.